tests/postgres_tests/test_array.py

```diff
@@ -414,6 +414,24 @@ def test_overlap_charfield_including_expression(self):
             [obj_1, obj_2],
         )
 
+    def test_overlap_values_array_field(self):
+        # issue 26056
+        post1 = CharArrayModel(field=["django"])
+        post2 = CharArrayModel(field=["thoughts"])
+        post3 = CharArrayModel(field=["tutorial"])
+        CharArrayModel.objects.bulk_create([post1, post2, post3])
+        qs = CharArrayModel.objects.filter(
+            field__overlap=CharArrayModel.objects.values_list("field", flat=True)
+        )
+        self.assertSequenceEqual(
```

Comment:
`qs` is not ordered so we should use `assertCountEqual()`:
```suggestion
        self.assertCountEqual(
```